pkg/secrets/secrets.go

```diff
@@ -0,0 +1,281 @@
+package secrets
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/containers/common/pkg/secrets/filedriver"
+	"github.com/containers/storage/pkg/lockfile"
+	"github.com/containers/storage/pkg/stringid"
+	"github.com/pkg/errors"
+)
+
+// maxSecretSize is the max size for secret data - 512kB
+const maxSecretSize = 512000
+
+// secretIDLength is the character length of a secret ID - 25
+const secretIDLength = 25
+
+// errInvalidPath indicates that the secrets path is invalid
+var errInvalidPath = errors.New("invalid secrets path")
```

Comment:
Can I ask why these aren't public? We usually do public errors in Podman-land so that `errors.Cause()` can be used to easily identify what went wrong.